### PR TITLE
PARO-41: Change Sonar config for reBuy default maven archetype

### PR DESCRIPTION
**Pull request by Sidorii**

### CODE REVIEW

File: src/main/resources/archetype-resources/Jenkinsfile [hunk 1]
```diff
@@ -2,4 +2,4 @@
 
 library('jenkins-libraries@v3.2.1')
```
Comment (der-eismann) [inline src/main/resources/archetype-resources/Jenkinsfile:3]:
```suggestion
library('jenkins-libraries@v3.3.0')
```
I just created a new release you can use
https://github.com/rebuy-de/jenkins-libraries/releases/tag/v3.3.0

File: src/main/resources/archetype-resources/Jenkinsfile [hunk 2]
```diff
@@ -2,4 +2,4 @@
...
 
-springBoot2Pipeline(jdk: 'jdk17', email: '${mailerRecipients}')
+springBoot2Pipeline(jdk: 'jdk17', qualityGateCheck: true, email: '${mailerRecipients}')
```
Comment (der-eismann) [inline src/main/resources/archetype-resources/Jenkinsfile:5]: Not sure why we didn't set it as default in the Jenkins job, but here

**Review by der-eismann — APPROVED** [review]

**Comment by der-eismann** [comment]
Does this check even work? I can't [see the stage anywhere](https://jenkins.rebuy.tools/blue/organizations/jenkins/rebuy-de%2Fparcel-routing-silo/detail/master/62/pipeline), what does it do?

File: src/main/resources/archetype-resources/Jenkinsfile (hunk 2 above)
Comment (Sidorii) [reply src/main/resources/archetype-resources/Jenkinsfile:5]: This property enables build failure when Sonar quality gate isn't passed. Most of our silos currently don't have proper configuration of sonar and have quality gate checks failed. So assigning this property here will affect only new silos that going to be created from this archetype without affecting old ones.

File: src/main/resources/archetype-resources/Jenkinsfile (hunk 1 above)
Comment (Sidorii) [reply src/main/resources/archetype-resources/Jenkinsfile]: Updated, thanks!

File: src/main/resources/archetype-resources/Jenkinsfile (hunk 2 above)
Comment (der-eismann) [reply src/main/resources/archetype-resources/Jenkinsfile:5]: :+1:

**Pull request merged**